streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java

```diff
@@ -51,22 +52,16 @@
     private final String logPrefix;
 
     private final Producer<byte[], byte[]> producer;
-    private final String applicationId;
```

Comment:
As we need to pass in `ConsumerGroupMetadata` on commitTx() for eos-beta, it's simpler to also pass in the `application.id` during commit to unify the logic.